clamav/cnudie.py

```diff
@@ -21,22 +25,103 @@
 ci.log.configure_default_logging()
 
 
+def _should_skip_scan(
+        resource_node: cnudie.iter.ResourceNode,
+        current_version_info: clamav.model.ClamAVVersionInfo,
+        virus_db_max_age_days: int,
+        skip_if_virus_definition_unchanged: bool,
+        delivery_client: delivery.client.DeliveryServiceClient = None,
+) -> bool:
+
+    if not delivery_client:
```

Comment:
rather mv this check to caller, do not allow passing-in non for delivery-service-client